src/mon/FSCommands.cc

```diff
@@ -968,6 +968,56 @@ class ResetFilesystemHandler : public FileSystemCommandHandler
   }
 };
 
+class RenameFilesystemHandler : public FileSystemCommandHandler
+{
+  public:
+  RenameFilesystemHandler()
+    : FileSystemCommandHandler("fs rename")
+  {}
+
+  int handle(
+      Monitor *mon,
+      FSMap& fsmap,
+      MonOpRequestRef op,
+      const cmdmap_t& cmdmap,
+      std::stringstream &ss) override
+  {
+    string fs_name;
+    cmd_getval(cmdmap, "fs_name", fs_name);
+    auto fs = fsmap.get_filesystem(fs_name);
+    if (fs == nullptr) {
+        ss << "filesystem '" << fs_name << "' does not exist";
+        // Unlike fs rm, we consider this case an error
+        return -ENOENT;
+    }
+
+    // check new filesystem doesn't already exist!
+    string new_fs_name;
+    cmd_getval(cmdmap, "new_fs_name", new_fs_name);
+    auto new_fs = fsmap.get_filesystem(new_fs_name);
+    if (!(new_fs == nullptr)) {
+        ss << "desired filesystem name '" << new_fs_name << "' already in use";
+        return -EINVAL;
+    }
+
+    // Check for confirmation flag
+    bool sure = false;
+    cmd_getval(cmdmap, "yes_i_really_mean_it", sure);
+    if (!sure) {
+      ss << "this is a potentially disruptive optionw, clients need to use IDs with access to new fs name.  "
+        "Add --yes-i-really-mean-it if you are sure you wish to continue.";
```

Comment:
```suggestion
      ss << "this is a potentially disruptive operation, clients cephx credentials need reauthorized to access the file system and its pools with the new name.  "
        "Add --yes-i-really-mean-it if you are sure you wish to continue.";
```

Reply:
Sorry! didn't mean to be tardy. I uploaded the draft just to show my progress.